osutil/disks/mapper.go

```diff
@@ -0,0 +1,64 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package disks
+
+import (
+	"fmt"
+	"os/exec"
+
+	"github.com/snapcore/snapd/osutil"
+)
+
+// CreateLinearMapperDevice creates a linear device mapping of the given device
+// with the given offset and size.
+//
+// The total size of underlying device must be offset+size, this is
+// not validated by this code.
+//
+// The mapper device node is returned.
+func CreateLinearMapperDevice(device, name, uuid string, offset, size uint64) (string, error) {
+	const dmSetupSectorSize = 512
```

Comment:
ah actually could you leave a TODO about this being used on 4k sector size devices? It seems dmsetup does not actually do any validation itself so if the device that is provided actually has a logical size of 4k, then things fall apart and a broken device is created: https://bugs.launchpad.net/ubuntu/+source/lvm2/+bug/1195980 though also that bug is 8 years old, but I don't have a 4k logical disk in front of me to verify if that's still the case or not

Reply:
added

Reply:
thank you